rclpy/src/rclpy/_rclpy.c

```diff
@@ -5494,6 +5506,24 @@ rclpy_publisher_get_topic_name(PyObject * Py_UNUSED(self), PyObject * args)
   return PyUnicode_FromString(topic_name);
 }
 
+
+/// Retrieves default domain id defined in rcl layer
+/**
+ *
+ * \return the defined domain id
+ */
+static PyObject *
+rclpy_get_default_domain_id(PyObject * Py_UNUSED(self), PyObject * args)
```

Comment:
i was expecting this would call https://github.com/ros2/rcl/blob/6f3c3fb10391edaacb2676e2345a754bef1b76f2/rcl/include/rcl/domain_id.h#L40-L42 to get default domain id from rcl.

Reply:
This function is called at  
https://github.com/Barry-Xu-2018/rclpy/blob/02d0bb6aa33b8e44a97ecac6cae4bbb8fce20f34/rclpy/rclpy/__init__.py#L63-L68. 
rcl_get_default_domain_id() may return error.  
rclpy_get_default_domain_id() directly return RCL_DEFAULT_DOMAIN_ID defined in rcl layer.  

Besides, this domain id (RCL_DEFAULT_DOMAIN_ID) will be set to init_options in rclpy_init() (_rclpy.c)  
In rcl_init(), if domain id equal to RCL_DEFAULT_DOMAIN_ID, rcl_get_default_domain_id() will be called.  
Refer to https://github.com/ros2/rcl/blob/6f3c3fb10391edaacb2676e2345a754bef1b76f2/rcl/src/rcl/init.c#L146-L154.   

So not directly call rcl_get_default_domain_id() here.

Reply:
that makes sense.